src/app/compare/compare.component.js

```diff
@@ -60,12 +61,24 @@ function CompareCtrl($scope, toaster, dimCompareService, dimStoreService, D2Stor
         }
       });
       vm.statsMap[statId] = itemStatsMap[statId].index;
-      vm.comparisons[0].stats.splice(vm.statsMap[statId], 0, { value: undefined, id: Number(statId), statHash: Number(statId), name: itemStatsMap[statId].name });
-    });
 
+      const missingStatItemIdx = vm.comparisons.findIndex((compItem) => !compItem.stats.some((stat) => stat.id === Number(statId)));
+      if (missingStatItemIdx > 0) {
+        vm.comparisons[missingStatItemIdx].stats.splice(vm.statsMap[statId], 0, { value: undefined, id: Number(statId), statHash: Number(statId), name: itemStatsMap[statId].name });
+      }
+    });
     return item;
   }
 
+  function removeMissingStats() {
+    _.map(vm.comparisons, (compItem) => {
+      const statIndex = compItem.stats.findIndex((stat) => (stat.bar && stat.base && stat.sort) === undefined);
```

Comment:
It's a bit more idiomatic to just say `!(stat.bar && stat.base && stat.sort)`. If you are nervous you can always use `Boolean(...)`.

Reply:
I actually more inclined to replace this including a "missingStat = true" and look for that instead of the absence of others. What do you think?

Reply:
oh, that'd be cool